tests/unit/lib/DynamicArray_unittest.cc

```diff
@@ -251,10 +252,44 @@ TEST_F(CDynamicArrayFixture, append_array_bool)
 	EXPECT_EQ(wrapper_array_b->get_element(2), true);
 }
 
-TEST_F(CDynamicArrayFixture, serializable_array)
+TYPED_TEST(CDynamicArrayTest, serializable_array)
 {
-	wrapper_array_i->save_serializable_pre();
-	wrapper_array_i->load_serializable_pre();
-	wrapper_array_b->save_serializable_pre();
-	wrapper_array_b->load_serializable_pre();
+	for (int32_t i = 0; i < 5; i++)
+	{
+		this->custom_array->push_back((TypeParam)CMath::random(0, 1));
+	}
+
+	/* generate file name */
+	char filename_tmp[] = "serialization-asciiCDynamicArray.XXXXXX";
+#ifdef _WIN32
+	int err = _mktemp_s(filename_tmp, strlen(filename_tmp) + 1);
+	ASSERT(err == 0);
+#else
+	int fd = mkstemp(filename_tmp);
+	ASSERT(fd != -1);
+	int retval = close(fd);
+	ASSERT(retval != -1);
+#endif
+	char* filename = filename_tmp;
+
+	/* create serialized copy */
+	CSerializableAsciiFile* file = new CSerializableAsciiFile(filename, 'w');
+	this->custom_array->save_serializable(file);
+	file->close();
+	SG_UNREF(file);
+
+	file = new CSerializableAsciiFile(filename, 'r');
+	CDynamicArray<TypeParam>* new_array = new CDynamicArray<TypeParam>();
+	new_array->load_serializable(file);
+	file->close();
+	SG_UNREF(file);
+
+	for (int32_t i = 0; i < this->custom_array->get_num_elements(); i++)
```

Comment:
first assert that the number of elements are the same, otherwise this might segfault
Use a variable for that (not just constant 5 in the code above)